### PR TITLE
Enable ICompileTypeAccessor on static compilation

### PR DESCRIPTION
**Pull request by twsouthwick**
This also hooks it up to enable Page.LoadControl(virtualPath)

### CODE REVIEW

File: test/Compiler.Dynamic.Tests/assets/testForUserControl/InsideTestUserControl.ascx
Comment (birojnayak) [inline test/Compiler.Dynamic.Tests/assets/testForUserControl/InsideTestUserControl.ascx:1]: why did you delete this ? this was testing user control inside user control.. it was a good test coverage...

File: samples/WebFormsSample/loadusercontrol.aspx.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+using System.Web.UI;
+using System.Web.UI.WebControls;
+using System.Web.UI.WebControls.WebParts;
+using System.Web.UI.HtmlControls;
+
+public partial class LoadUserControl : System.Web.UI.Page 
+{
+    protected void Page_Load(object sender, EventArgs e)
+    {
+        panel1.Controls.Add(Page.LoadControl("~/TestUserControl.ascx"));
```
Comment (birojnayak) [inline samples/WebFormsSample/loadusercontrol.aspx.cs:15]: nice :)

File: test/Compiler.Dynamic.Tests/assets/testForUserControl/InsideTestUserControl.ascx
Comment (twsouthwick) [reply test/Compiler.Dynamic.Tests/assets/testForUserControl/InsideTestUserControl.ascx:1]: It's duplicated - the tests were moved to folder  test 09 and those are being run. This doesn't change test coverage

File: src/Compiler.Dynamic/SystemWebCompilation.cs [hunk 2]
```diff
@@ -31,36 +31,30 @@ internal abstract class SystemWebCompilation<T> : IDisposable
 
     private SystemWebCompilationUnit _compiled = new();
 
-    private sealed class SystemWebCompilationUnit : ICompiledTypeAccessor
+    private sealed class SystemWebCompilationUnit : ICompiledTypeAccessor, IDisposable
```
Comment (birojnayak) [inline src/Compiler.Dynamic/SystemWebCompilation.cs:34]: make sense

**Review by birojnayak — COMMENTED** [review]
some clarifications

**Review by birojnayak — APPROVED** [review]
